pkg/cmd/opts/helm.go

```diff
@@ -661,7 +661,7 @@ func (o *CommonOptions) HelmInitDependencyBuild(dir string, chartRepos []string,
 	// lets stick with helm2 for this step
 	//
 	helmBinary := o.Helm().HelmBinary()
-	o.Helm().SetHelmBinary("helm")
+	o.Helm().SetHelmBinary(helmBinary)
```

Comment:
What's about the comment here? It seems it was an explicit choice to use "helm". Is this not an issue anymore?

Reply:
I didn't see any problem, using helm3, in this step, maybe I'm missing something, but for me it works... I missed the comment, does removing it work for you?

Reply:
I think we would need to remove more than just the comment. As the code is written now it is a noop and can effectively be removed, right:

```
helmBinary := o.Helm().HelmBinary()
o.Helm().SetHelmBinary(helmBinary)
```

This gets the helm binary from the `Helmer` and then set the very value it just retrieved. So really if we want to do anything here it would be removing the comment + the two lines above. The explicit setting of "helm" was a workaround when using helm3 which you are saying is not needed anymore.

Reply:
I think you are right.
For all future changes we should adopt the following though:

The way I understand the code is that helmBinary is the name of the executable to be used when executing the command, this is needed because in our docker image we have helm and helm3 (the two binaries), normally users (outside the pipeline) have on "helm" and call it directly. helm2 and helm3 differ in few way:
- no helm init for helm3
- --name is remove in helm3 and is part of the params
...
In theory we should always call helm but in the pipeline the JX_HELM3 param should be used to tell use to set helmBinary to helm3.
Whenever we check which helm version we are using before running a command, instead of checking against helmBinary we should check against helm.version.

I will first fix this PR, but someone needs to be attributed this task to fix all I just explained...